src/fields/elements/getContentTypeElements.ts

```diff
@@ -0,0 +1,27 @@
+import { ZObject } from 'zapier-platform-core';
+import { KontentBundle } from '../../types/kontentBundle';
+import { getContentType } from '../../utils/types/getContentType';
+import { ContentTypeElements, ManagementClient } from '@kontent-ai/management-sdk';
+import { createManagementClient } from '../../utils/kontentServices/managementClient';
+
+export const getContentTypeElements = (z: ZObject, bundle: KontentBundle<{}>, contentTypeId: string) => {
+  if (!contentTypeId) {
+    return Promise.resolve([]);
+  }
+  const managementClient = createManagementClient(z, bundle);
+
+  return getContentType(z, bundle, contentTypeId)
+    .then(type => Promise.all(type.elements
+      .map(el => el.type === 'snippet' ? getSnippetElements(managementClient, el) : Promise.resolve([el])))
+      .then(allElements => allElements.flat()));
+};
+
+const getSnippetElements = (managementClient: ManagementClient, element: ContentTypeElements.ISnippetElement) =>
+  managementClient
+    .viewContentTypeSnippet()
+    .byTypeId(element.snippet.id ?? '')
+    .toPromise()
+    .then(res => res.data.elements.map(el => ({
+      ...el,
+      codename: `${res.data.codename}__${(el as Exclude<typeof el, ContentTypeElements.ISnippetElement>).codename}`, // type override is because of MAPI SDK bad types, snippet cannot contain snippet element
```

Comment:
It explains why I have to use the `as ...` override. It has two reasons that combine together here.

1.  The MAPI SDK TS types don't contain the `codename` property on the `ISnippetElement` even though it should have it. (I already created an issue for that https://github.com/kontent-ai/management-sdk-js/issues/90) 
2.  The content type snippet elements cannot contain `ISnippetElement` even though the MAPI SDK TS types claim they can. Thanks to this fact I can ignore the first issue.

I am also considering filing another issue for the second problem. However, I am not sure about it, because even the MAPI documentation is wrong.